careers/software-engineer.md

```diff
@@ -44,6 +73,21 @@ An engineer II is capable of taking on most technical problems and coordinating
 
 **Collaboration:** They are mindful of the work that others are doing and can critique or suggest improvements to the proposed plans of others. They get help from others when stuck, but are also capable of helping unstick their peers. They are capable of effectively running small to medium-sized projects with regular coaching.
 
+### Responsibilities
+
+* Design and develop software components or features with moderate complexity.
+* Write clean, efficient, and well-documented code with less direct supervision.
+* Perform more thorough software testing, including unit and integration tests.
```

Comment:
This is pedantic, but saying "more thorough software testing" and then following it up with unit testing reads a a bit off.

Instead of implying a progression in terms of how testing strategies are used, maybe we could try something like:

> Demonstrate sound judgment in testing strategies, selecting appropriate testing methods to ensure feature/component quality while maintaining development velocity.